stubs/fpdf2/fpdf/fpdf.pyi

```diff
@@ -55,25 +61,31 @@ class SubsetMap:
 def get_page_format(format: _Format | tuple[float, float], k: float | None = ...) -> tuple[float, float]: ...
 def load_cache(filename: Path): ...
 
+# TODO: TypedDicts
+_Page = dict[str, Any]
+_Font = dict[str, Any]
+_FontFile = dict[str, Any]
+_Image = dict[str, Any]
+
 class FPDF:
     MARKDOWN_BOLD_MARKER: str
     MARKDOWN_ITALICS_MARKER: str
     MARKDOWN_UNDERLINE_MARKER: str
-    offsets: Any
+    offsets: dict[int, int]
     page: int
     n: int
-    buffer: Any
-    pages: Any
-    state: Any
-    fonts: Any
-    font_files: Any
-    diffs: Any
-    images: Any
-    annots: Any
-    links: Any
+    buffer: bytearray
```

Comment:
`__init__` has `self.buffer=''`, so this can also be a string

Reply:
Where do you see that line? I can only see this initialization:

https://github.com/PyFPDF/fpdf2/blob/07ebb46d690cccf7449dbb37fdb2ec4fba4d3d14/fpdf/fpdf.py#L255

The old fpdf package (without the 2) has this line:

https://github.com/reingart/pyfpdf/blob/cb3340806c140b6e63eb4ce9d6230be84fba456e/fpdf/fpdf.py#L67

Reply:
Sorry, I was looking at fpdf instead of fpdf2.